### PR TITLE
gh-105522: Correctly handle possible exceptions in `Enum` creation in `test_enum`

### PR DESCRIPTION
**Pull request by sobolevn**
This is simplier now! 🎉 

* Issue: gh-105522

### CODE REVIEW

**Comment by sobolevn** [comment]
Here's how a potential exception looks like:

```
======================================================================
ERROR: test_pickle_enum (test.test_enum.TestSpecial.test_pickle_enum)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sobolev/Desktop/cpython/Lib/test/test_enum.py", line 1458, in test_pickle_enum
    raise Stooges
  File "/Users/sobolev/Desktop/cpython/Lib/test/test_enum.py", line 65, in <module>
    class Stooges(Enum):
  File "/Users/sobolev/Desktop/cpython/Lib/enum.py", line 494, in __prepare__
    raise RuntimeError('Exc', cls)
RuntimeError: ('Exc', 'Stooges')
```

And here's how it looks if there are multiple exceptions:

```
======================================================================
ERROR: test_floatenum_fromhex (test.test_enum.TestSpecial.test_floatenum_fromhex)
----------------------------------------------------------------------
  + Exception Group Traceback (most recent call last):
  |   File "/Users/sobolev/Desktop/cpython/Lib/test/test_enum.py", line 55, in inner
  |     raise ExceptionGroup('Enum Exceptions', excs)
  | ExceptionGroup: Enum Exceptions (2 sub-exceptions)
  +-+---------------- 1 ----------------
    | Traceback (most recent call last):
    |   File "/Users/sobolev/Desktop/cpython/Lib/test/test_enum.py", line 81, in <module>
    |     class FloatStooges(float, Enum):
    |   File "/Users/sobolev/Desktop/cpython/Lib/enum.py", line 494, in __prepare__
    |     raise RuntimeError('Exc', cls)
    | RuntimeError: ('Exc', 'FloatStooges')
    +---------------- 2 ----------------
    | Traceback (most recent call last):
    |   File "/Users/sobolev/Desktop/cpython/Lib/test/test_enum.py", line 108, in <module>
    |     class IntFlagStooges(IntFlag):
    |   File "/Users/sobolev/Desktop/cpython/Lib/enum.py", line 494, in __prepare__
    |     raise RuntimeError('Exc', cls)
    | RuntimeError: ('Exc', 'IntFlagStooges')
    +------------------------------------
```

**Comment by sobolevn** [comment]
@ethanfurman friendly ping (x2) 😄

File: Lib/test/test_enum.py [hunk 1]
```diff
@@ -38,6 +38,25 @@ def load_tests(loader, tests, ignore):
                 ))
     return tests
 
+def reraise_enum_exceptions(*enum_types_or_exceptions):
```
Comment (ethanfurman) [inline Lib/test/test_enum.py]:
```suggestion
def reraise_if_not_enum(*enum_types_or_exceptions):
```

**Review by ethanfurman — CHANGES_REQUESTED** [review]
One minor change and it looks good!

File: Lib/test/test_enum.py (hunk 1 above)
Comment (sobolevn) [reply Lib/test/test_enum.py]: Done! 👍

**Comment by sobolevn** [comment]
I have made the requested changes; please review again

**Review by ethanfurman — APPROVED** [review]

**Pull request merged**